### ISSUE TITLE
JSDoc `@callback` tag types are only checked if referenced by code

### ISSUE THREAD

**Issue by michaelfig** [microsoft/TypeScript#47608]
# Bug Report

### 🔎 Search Terms

`@callback` jsdoc "no error"

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about `@callback`

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?filetype=js#code/PTAEAEBcGcFoGMAWBTeBrAsAKG8AVHtqHhPAIYA2FARmeqAKoB2ATsgGbJtPzIAmAYWpESAEQD2yaKCbjIoNgAdxLeWSaguLFaHEaARLT6QAnouT6AdCIhtIAVxZNpAbyOnzAXxHBsuAjbg5FS09ABKHFzIPPxCNhHKqtLqmizaLLoG7mYWADSg1PbyehQmoACW7KARnNy8gtQV0my10fUFZXQOlKDw4nzI1ljEtsgOTq7ZXqBk1PA+flj4hMMkUDkDVW5kxjmejKyRdfwAKjk2AAra5iylCsiJMDMaWjp6oIY7HhZDxL44SwIEG+oBcNSiMQa+wA6uVIIhxEVQPDys0jm1kPl4cgCl8chUNODjg0mhUAOayNh8Sx-bB9ZzydjicSgAC8oAAFABKNkAPlAAHJZvABQBubBAA)

### 💻 Code

```ts
// @ts-check

/**
 * @callback UnreferencedCb
 * Does not report an error on "badtype".
 * @returns {badtype}
 */

/**
 * @callback ReferencedCb
 * Reports an error on "badtype", but only if ReferencedCb is referenced by actual code.
 * @returns {badtype} abc
 */

/**
 * @typedef {badtype} UnreferencedType
 * Properly reports an error on "badtype".
 */

/** @type {ReferencedCb} Without this reference, the badtype in ReferencedCb is ignored. */
const foo = () => 'abc';
```

### 🙁 Actual behavior

The `badtype` in the definition `UnreferencedCb` is not marked as an unknown name, but it is one.  Typing some JS code with `UnreferencedCb` causes TS to behave correctly and be marked as an unknown name.

The other `badtype`s are correctly marked as unknown names (`Cannot find name 'badtype'.(2304)`).

### 🙂 Expected behavior

All occurrences of `badtype` should be marked as syntax errors with `Cannot find name 'badtype'.(2304)`, since that type is not defined.

The `@callback` JSDocs should be checked when `// @ts-check` is enabled, regardless of whether they are actually referenced by JS code in the same file.

**Comment by RyanCavanaugh** [comment]
This is fixed in the current native TypeScript implementation. With `--allowJs --checkJs`, TypeScript 7.1.0-dev.20260908.1 reports TS2304 for all three `badtype` references, including the unreferenced `UnreferencedCb` return type. Classic TypeScript 6.0.3 still reports only the referenced callback and typedef cases.
